include/xtensor/xreducer.hpp

```diff
@@ -572,34 +705,45 @@ namespace xt
      */
 
     template <class F, class E, class X, class EVS = DEFAULT_STRATEGY_REDUCERS,
-              XTL_REQUIRES(xtl::negation<is_evaluation_strategy<X>>,
+              XTL_REQUIRES(xtl::negation<is_reducer_options<X>>,
                            detail::is_xreducer_functors<F>)>
-    inline auto reduce(F&& f, E&& e, X&& axes, EVS evaluation_strategy = EVS())
-    {
-        return detail::reduce_impl(std::forward<F>(f), std::forward<E>(e), std::forward<X>(axes), evaluation_strategy);
+    inline auto reduce(F&& f, E&& e, X&& axes, EVS /*evaluation_strategy*/ = EVS())
+    {
+        return detail::reduce_impl(std::forward<F>(f),
+                                   std::forward<E>(e),
+                                   std::forward<X>(axes),
+                                   typename reducer_options<EVS>::evaluation_strategy{},
+                                   typename reducer_options<EVS>::keep_dims{}
+        );
     }
 
     template <class F, class E, class X, class EVS = DEFAULT_STRATEGY_REDUCERS,
-              XTL_REQUIRES(xtl::negation<is_evaluation_strategy<X>>,
+              XTL_REQUIRES(xtl::negation<is_reducer_options<X>>,
                            xtl::negation<detail::is_xreducer_functors<F>>)>
     inline auto reduce(F&& f, E&& e, X&& axes, EVS evaluation_strategy = EVS())
     {
-        return reduce(make_xreducer_functor(std::forward<F>(f)), std::forward<E>(e), std::forward<X>(axes), evaluation_strategy);
+        return reduce(make_xreducer_functor(std::forward<F>(f)), std::forward<E>(e), 
+                      std::forward<X>(axes), evaluation_strategy);
     }
 
     template <class F, class E, class EVS = DEFAULT_STRATEGY_REDUCERS,
-              XTL_REQUIRES(is_evaluation_strategy<EVS>,
+              XTL_REQUIRES(is_reducer_options<EVS>,
                            detail::is_xreducer_functors<F>)>
-    inline auto reduce(F&& f, E&& e, EVS evaluation_strategy = EVS())
+    inline auto reduce(F&& f, E&& e, EVS /*evaluation_strategy*/ = EVS())
     {
         xindex_type_t<typename std::decay_t<E>::shape_type> ar;
         resize_container(ar, e.dimension());
         std::iota(ar.begin(), ar.end(), 0);
-        return detail::reduce_impl(std::forward<F>(f), std::forward<E>(e), std::move(ar), evaluation_strategy);
+        return detail::reduce_impl(std::forward<F>(f),
+                                   std::forward<E>(e),
+                                   std::move(ar),
+                                   typename reducer_options<EVS>::evaluation_strategy{},
+                                   typename reducer_options<EVS>::keep_dims{}
```

Comment:
It looks like `keep_dims` is always passed, how can the user specify another value?

Reply:
in this case, keep dims is statically parsed from the EVS options tuple. we can rename this to `is_keep_dims` to make it clearer?